syncer/filter.go

```diff
@@ -112,13 +55,50 @@ func (s *Syncer) skipDMLEvent(table *filter.Table, eventType replication.EventTy
 	case replication.DELETE_ROWS_EVENTv0, replication.DELETE_ROWS_EVENTv1, replication.DELETE_ROWS_EVENTv2:
 		et = bf.DeleteEvent
 	default:
-		return false, terror.ErrSyncerUnitInvalidReplicaEvent.Generate(eventType)
+		filtered, err := s.filterOneEvent(table, bf.NullEvent, "")
+		if err != nil {
+			return false, err
+		} else if filtered {
+			return filtered, nil
+		} else {
+			return false, terror.ErrSyncerUnitInvalidReplicaEvent.Generate(eventType)
+		}
 	}
+	return s.filterOneEvent(table, et, "")
+}
 
-	action, err := s.binlogFilter.Filter(schemaName, tableName, et, "")
+// filterSQL filter unsupported sql in tidb and global sql-patterns in binlog-filter config file.
+func (s *Syncer) filterSQL(sql string) (bool, error) {
+	if utils.IsBuildInSkipDDL(sql) {
+		return true, nil
+	}
+	action, err := s.binlogFilter.Filter("", "", bf.NullEvent, sql)
 	if err != nil {
-		return false, terror.Annotatef(terror.ErrSyncerUnitBinlogEventFilter.New(err.Error()), "skip row event %s on `%v`", eventType, table)
+		return false, terror.Annotatef(terror.ErrSyncerUnitBinlogEventFilter.New(err.Error()), "skip query %s", sql)
 	}
+	return action == bf.Ignore, nil
+}
 
+// filterOneEvent will return true when
+// - given `sql` matches builtin pattern.
+// - any schema of table names is system schema.
+// - any table name doesn't pass block-allow list.
+// - type of SQL doesn't pass binlog filter.
+// - pattern of SQL doesn't pass binlog filter.
+func (s *Syncer) filterOneEvent(table *filter.Table, et bf.EventType, sql string) (bool, error) {
+	if filter.IsSystemSchema(table.Schema) {
```

Comment:
seems not contains `matches builtin pattern` check?

Reply:
What is builtin pattern check?

Reply:
in this func's comment 
```
// skipOneEvent will return true when
// - given `sql` matches builtin pattern.
```

Reply:
> seems not contains matches builtin pattern check?

you are right.

> // - given `sql` matches builtin pattern.

this represents `utils.IsBuildInSkipDDL`. I will delete this comment next commit.